adr/0007-use-jwt-tokens-as-authentication-mechanism.md

```diff
@@ -0,0 +1,35 @@
+# 7. Use JWT tokens as authentication mechanism
+
+Date: 2022-12-06
+
+## Status
+
+Accepted
+
+## Context
+
+We have cookie based authentication in place on the trento-web project. The decision was made based on previous requirements but with the evolution of the `trento platform` we need a different authentication method.
+The authentication method should be cookie-less, stateless and capable of authenticating more services with a single authentication token.
+
+The new service, `wanda`, responsible for the new check engine execution, needs an authentication mechanism.
+
+The new authentication solution should be based on standards and should be easy and appropriate to integrate into both Single Page Applications and raw API consumption.
+
+## Decision
+
+Decided on [JWT](https://jwt.io) token-based authentication.
+
+We don't want to use an API gateway and an external identity provider because the effort of deploying and mantaining these new pieces of architecture does not make sense right now.
+
+We want to use the web dashboard as an identity provider and authentication coordinator, this is a good compromise between the level of security we need and the type of application we are building right now. The choice of jwt as a token standard implies that a future switch to an external identity provider and/or API gateway will be a painless and easy procedure for our applications. 
+
```

Comment:
let's go a little bit more in detail with the considered alternatives if possible

Reply:
done